articles/060_ros_middleware_interface.md

```diff
@@ -353,6 +339,11 @@ The described concept has been implemented in the following packages:
     - the DDS specific code based on IDL files for each message
     - additional code to enable invoking the register/create/convert/write functions for each message type
 
+- the package [rmw_implementation](https://github.com/ros2/rmw_implementation/tree/master/rmw_implementation) provides the mechanism to switch between compile time and runtime selection of the middleware implementation
+
+  - if only one implementation is available at compile time it links directly against it
+  - if multiple implementations are available it implements the middleware interface itself and acts according to the strategy pattern by loading the shared library of a specific middleware implementation identified by an environment variable at runtime and pass all calls along
```

Comment:
feel free to discard if you think it's too repetitive "if multiple implementations are available **at compile time**"

Reply:
Added in 9ada2feede26253276287e533020ad1659a9695f.